### DISCUSSION TITLE
Processing of TCP packets sent by the SYSTEM process using windows-redirector.exe

### DISCUSSION THREAD

**Discussion by sbromt** [mitmproxy/mitmproxy#7375] [Q&A]
When accessing the NAS through Windows Explorer, the TCP handshake packets from the System process (PID=4) are held at the if block on line 188 of the main2.rs. However, when a socket event occurs, processing is skipped at the if block on line 232 of the main2.rs, preventing Windows Explorer from establishing a TCP connection.

Wouldn't it be better to reinject the connections held in connections back into the network before skipping them, even if the process is the System process?

**Comment by Prinzhorn** [comment]
This sounds like it should be a bug report? Maybe you can open an issue with more details and also link to the corresponding code in the repo. Rust bits are here: https://github.com/mitmproxy/mitmproxy_rs (but open a report here)

Thanks!

**Reply by sbromt** [reply]
Thank you for your response.
I couldn’t find the "Issues" menu on the mitmproxy_rs page you mentioned.
How can I submit a bug report?

**Reply by Prinzhorn** [reply]
In this repo, here's the direct link https://github.com/mitmproxy/mitmproxy/issues/new?assignees=&labels=kind%2Ftriage&projects=&template=bug_report.md&title=

**Reply by sbromt** [reply]
Thank you. I will post it there.

**Reply by sbromt** [reply]
I have reposted below. Thank you for your assistance.

https://github.com/mitmproxy/mitmproxy/issues/7376

**Reply by Prinzhorn** [reply]
Thanks for opening the issue https://github.com/mitmproxy/mitmproxy/issues/7376, I'm going to close this discussion.
